### PR TITLE
Feature/ele 596 empty textboxes are not rendered no placeholder shown

### PR DESCRIPTION
**Pull request by dannelundqvist**
Major improvements to useYValue() which enables us to (almost) automatically create missing structures like tt/slugline which is often missing. I think we now have all the pieces to replace useYObserver() completely.

### CODE REVIEW

File: src/hooks/useYValue.tsx [hunk 1]
```diff
@@ -31,13 +43,28 @@ export function useYValue<T>(path: string): [
...
+      setValue(isYXmlText(vp[0]) ? vp[0].toJSON() as T : vp[0] as T)
+    } else {
+      // Just return raw Y values when not observing
+      setValue(vp[0] as T)
+    }
 
-    if (isYContainer(p)) {
-      setParent(p)
+    if (isYContainer(vp[1])) {
+      setParent(vp[1])
```
Comment (gularsson) [inline src/hooks/useYValue.tsx:65]: I'm not too found of this, I'd rather see something along the line as the previous (and even more verbose):
`const [value, path] = getValueByPath(...)`

No one char variables, and no magic numbers.

Though I see now that you're reassigning it, which kind makes it harder.
I'm not too found of reassigning either :wink:

**Review by gularsson — APPROVED** [review]
One thing to consider

File: src/hooks/useYValue.tsx (hunk 1 above)
Comment (dannelundqvist) [reply src/hooks/useYValue.tsx]: Good points. I Agree. I'll refactor it!

**Review by Victor-Lindh — APPROVED** [review]

**Review by gularsson — APPROVED** [review]
:+1:

**Pull request merged**